File: app/src/main/java/it/paranoidsquirrels/idleguildmaster/storage/data/quests/instances/TabulaRasa.java
package it.paranoidsquirrels.idleguildmaster.storage.data.quests.instances;

import it.paranoidsquirrels.idleguildmaster.R;
import it.paranoidsquirrels.idleguildmaster.storage.data.quests.Quest;
import it.paranoidsquirrels.idleguildmaster.storage.data.quests.QuestsManager;

/* JADX INFO: loaded from: classes3.dex */
public class TabulaRasa extends Quest {
    @Override // it.paranoidsquirrels.idleguildmaster.storage.data.quests.Quest
    public void calculateTargetProgress(int i) {
        this.targetProgress = ((long) (i - 2)) * 10;
    }

    @Override // it.paranoidsquirrels.idleguildmaster.storage.data.quests.Quest
    protected void configure() {
        this.idName = R.string.quest_tabula_rasa_name;
        this.idDescription = R.string.quest_tabula_rasa_description;
        this.defaultRarity = 3;
        this.minimumDifficulty = 3;
    }

    @Override // it.paranoidsquirrels.idleguildmaster.storage.data.quests.Quest
    public void realignStaticReference() {
        QuestsManager.tabulaRasa = this;
    }
}

File: app/src/main/kotlin/it/paranoidsquirrels/idleguildmaster/storage/data/quests/instances/TabulaRasa.kt
package it.paranoidsquirrels.idleguildmaster.storage.data.quests.instances

import it.paranoidsquirrels.idleguildmaster.R
import it.paranoidsquirrels.idleguildmaster.storage.data.quests.Quest
import it.paranoidsquirrels.idleguildmaster.storage.data.quests.QuestsManager

class TabulaRasa : Quest() {
    override fun calculateTargetProgress(difficulty: Int) {
        targetProgress = ((difficulty - 2).toLong() * 10).toLong()
    }

    override fun configure() {
        idName = R.string.quest_tabula_rasa_name
        idDescription = R.string.quest_tabula_rasa_description
        defaultRarity = 3
        minimumDifficulty = 3
    }

    override fun realignStaticReference() {
        QuestsManager.tabulaRasa = this
    }
}
